assets/js/base/components/chip/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * Component used to render a "chip" -- that is a list of inline items, each
+ * containing some text with an X button to remove/dismiss each chip.
+ *
+ * Each chip defaults to a list element but this can be customized by providing
+ * a wrapperElement.
+ */
+const Chip = ( {
+	text,
+	element = 'li',
+	className = '',
+	onRemove = () => {},
```

Comment:
Might be a tiny bit clearer to add "click" to this prop name so it's obvious, e.g. `onClickRemove` or `onRemoveClick`. Not a blocker :)

Reply:
Our current chip in the `dropdown-selector` component uses `onRemoveItem`. Maybe the naming is like this in case things ever are removed without a click; onRemove is the event and the event is what the consumer of this component cares about.

Reply:
The dropdown selector is a bit special because depending on the mode you can remove chips clicking on them or clicking on the cross button, there is also the possibility to remove them with the <kbd>Backspace</kbd>. So `click` would not make sense there.

For me, all suggestions you said would work in this case, so in my opinion I think we can merge this as-is and later iterate on it if we find another name for the prop would be more appropriate.